src/EditorFeatures/Core/Implementation/CodeCleanup/CodeCleanupService.cs

```diff
@@ -0,0 +1,165 @@
+                new[] { IDEDiagnosticIds.InlineDeclarationDiagnosticId });
+            dictionary.Add(FeatureOnOffOptions.RemoveUnusedVariables,
+                new[] { "CS0168", "CS0219" });
+            dictionary.Add(FeatureOnOffOptions.FixObjectCollectionInitialization,
+                new[] { IDEDiagnosticIds.UseObjectInitializerDiagnosticId, IDEDiagnosticIds.UseCollectionInitializerDiagnosticId });
+            //dictionary.Add(FeatureOnOffOptions.FixLanguageFeatures,
+            //    new[] { IDEDiagnosticIds. });
+            //IDEDiagnosticIds.PreferIntrinsicPredefinedTypeInDeclarationsDiagnosticId,
+            //    IDEDiagnosticIds.PreferIntrinsicPredefinedTypeInMemberAccessDiagnosticId,
+            //    IDEDiagnosticIds.InlineAsTypeCheckId,
+            //    IDEDiagnosticIds.InlineIsTypeCheckId,
+            //    IDEDiagnosticIds.InlineIsTypeWithoutNameCheckDiagnosticsId,
+
+            return dictionary.ToImmutableDictionary();
+        }
+
+        public Task<IEnumerable<TextChange>> GetChangesForCleanupDocument(Document document, CancellationToken cancellationToken)
+        {
+            var oldDocument = document;
+            document = RemoveSortUsings(document, cancellationToken);
+            document = ApplyCodeFixes(document, cancellationToken);
+
+            return document.GetTextChangesAsync(oldDocument, cancellationToken);
+        }
+        private Document RemoveSortUsings(Document document, CancellationToken cancellationToken)
+        {
+            // remove and sort usings
+            if (document.Project.Solution.Workspace.Options.GetOption(FeatureOnOffOptions.RemoveUnusedUsings, LanguageNames.CSharp))
+            {
+                var removeUsingsService = document.GetLanguageService<IRemoveUnnecessaryImportsService>();
+                if (removeUsingsService != null)
+                {
+                    document = removeUsingsService.RemoveUnnecessaryImportsAsync(document, cancellationToken).WaitAndGetResult(cancellationToken);
+                }
+            }
+
+            // sort usings
+            if (document.Project.Solution.Workspace.Options.GetOption(FeatureOnOffOptions.SortUsings, LanguageNames.CSharp))
+            {
+                document = OrganizeImportsService.OrganizeImportsAsync(document, cancellationToken).WaitAndGetResult(cancellationToken);
+            }
+
+            return document;
+        }
+
+        private Document ApplyCodeFixes(Document document, CancellationToken cancellationToken)
+        {
+            var fixAllService = document.Project.Solution.Workspace.Services.GetService<IFixAllGetFixesService>();
+
+            var dummy = new ProgressTracker();
+            foreach (var diagnosticId in GetEnabledDiagnosticIds(document.Project.Solution.Workspace))
+            {
+                var length = document.GetSyntaxTreeAsync(cancellationToken).WaitAndGetResult(cancellationToken).Length;
```

Comment:
why not make this method async?  blocking here seems very bad to me.

Reply:
fixed